### ISSUE TITLE
Intellisense's 'Suggest selection' should keep working when you have typed something

### ISSUE THREAD

**Issue by natanfudge** [microsoft/vscode#66019]
VSCode has a feature that remembers recent completions the user has selected and then automatically choosen those completions for them the next time. 
The problem is that it behaves quite oddly, and here is an example that illustrates why:

- Create a new Typescript file.
- Type `console.` vscode will suggest all of the available methods of `console`.
 - Choose `exception`.
- Type `console.` again. Vscode will immediately suggest `exception`, as expected.
And now here is the weird part: Type `e`, and instead of still suggesting `exception`, it will suggest `error`!

The problem here is that once you type anything after the suggestion came, vscode will completely disregard your completion history and will just give you the best match lexicographically. Even if in the example you have completed `exception` by typing an `e` and then having it complete it, the same thing will occur.
My suggestion is that when you have already typed something, the 'suggest selection' will just be filtered by what you have typed, instead of stop working. So in the example, after you have typed `e`, it will still suggest `exception`.

_"Switch the Suggest Selection settings to "recentlyUsedByPrefix" and it will work!"_
Yes it does make it sometimes work but overall it makes the situation much worse. For example if you type just `console.` it won't suggest exception no matter what you have done before.

**Comment by natanfudge** [comment]
Thanks for the response jrieken. 
When typing `l`, I would have expected it to complete `log` because _`exception` has already been filtered out._  On the other hand, when I type `e` It's pretty obvious I want to type `exception` again, since `exception` starts with `e`. Although when I type `er`, exception should absolutely not be an option, and `error` should again be the dominant suggestion. 

Another possible scenario is this: I first type `console.` and complete to `time`. After this, when I type `console.i`, technically `time` was not filtered out. Here there are two options: we either give `time` priority since it was typed before and not filtered, or we give `info` priority since it matches better. Here I could see why you would give `info` priority since it's a stretch to say that when I type `i` I mean to type `time`. 
Although, in the example I showed (with the error/exception), `exception` is a very good match, and so should definitely should still be suggested, and not `error`. 

So to sum up my opinion:
 - When there is a very good match (for example `e->exception`) the suggestion should definitely keep suggestion the same thing (`exception` in the example).
 - When there is no match (for example `l->exception`) the suggestion should definitely be ignored (so complete to log).
- When it matches but not perfectly (for example `i->time`), suggesting `time` is debatable, and I'm leaning towards yes, but I can see why that would seem like "you can't type anything else".

**Comment by jrieken** [comment]
Ok, that's kind of what it does today "Select recent suggestions unless further typing selects one". The thing is that `e` and `error` or `e` and `exception` score equally well and that then the original order (alphabetical) is used to make a decision. I can check if we can use the history instead.
